### ISSUE TITLE
Anaconda Launcher fatal error in Ubuntu 14.04 LTS

### ISSUE THREAD

**Issue by lloydbrodsky** [ContinuumIO/anaconda-issues#235]
Error message is:
Error: spawn ENOENT
    at exports._errnoException (util.js:742:11)
    at Process.ChildProcess._handle.onexit (child_process.js:1027:32)
    at child_process.js:1110:20
    at process._tickCallback (node.js:279:11)

Has been working before. I'm using Python 3.4 and 64 bit Ubuntu. Did try updating both the system and with conda.

**Comment by evplatt** [comment]
Same error here on 14.04.01 LTS 64bit.  Fresh Anaconda install (first time install) and installed launcher package.

**Comment by lazygun37** [comment]
Same error here, but with python 2.7. Also using Ubuntu 14.04 LTS

**Comment by tswicegood** [comment]
Hey guys!  I'm actively working on a new version of Launcher that will fix these issues.  No exact ETA yet, but shooting to have something out within the next few weeks.  I'll post here when I have a beta release ready if y'all are interested in helping test it out.

**Comment by syko4MT** [comment]
I hate to say, but the same error still exists. I just did clean installs of:
  - LinuxMint 17.2 MATE-64
  - Anaconda3-2.3.0-Linux-x86_64

on my hp4530s laptop; i3 2ndGen, 8G RAM, intel's grfx, 250G HDD.

I'm attempting to start the launcher script from the ~/Anaconda3/bin dir. My error message is
exactly the same as Lloyds.

But you still do awesome work!  ;-)

**Comment by iJonLuu** [comment]
Ubuntu 14.04TS:  same error:

Error: spawn ENOENT
    at exports._errnoException (util.js:742:11)
    at Process.ChildProcess._handle.onexit (child_process.js:1027:32)
    at child_process.js:1110:20
    at process._tickCallback (node.js:279:11)

Is a new version of launcher still coming real soon now ? ;-) :-)
Thanks

**Comment by goanpeca** [comment]
It is coming soon!

We will post in the mailing list so people can give it a try :-)

**Comment by alaindomissy** [comment]
Great! 
Is the current node-webkit based launcher going to be replaced by this PyGt project ? 
https://github.com/spyder-ide/conda-manager

**Comment by goanpeca** [comment]
The next version of the launcher is a nw.js project (so still node webkit).

The PyQt project you reference (*spolier... I am the developer...*) is separate project aiming to include a conda package manager within spyder via the plugin interface.

**Comment by roblem** [comment]
As a temporary workaround, I was able to successfully launch launcher on Fedora 22 using this script:

```
#!/bin/bash
source $HOME/anaconda/bin/activate ~/anaconda
launcher
```

One can then add a desktop launcher in Gnome with 
```
[Desktop Entry]
Type=Application
Name=Anaconda Launcher
Icon=/path/to/icon
Exec=/path/to/script/referenced/above
Terminal=true
Categories=Development;Research;
```
by saving in ~/.local/share/applications as `launcher.desktop`

**Comment by sandipank** [comment]
Same issue here...Please notify as soon as the problem is sorted out... thanks in advance

**Comment by alaindomissy** [comment]
Yes Please!!!
I am running a study group every Saturday
We recommend and install Anaconda for new python users multiple times every
week
It is a big pain that launcher is not working prperly for linux
I f you are not going to fix it soon, please open source it!
Don loose the good will you have in this community!!
Thanks

On Sat, Sep 26, 2015 at 1:49 AM, sandipank <notifications@github.com> wrote:

> Same issue here...Please notify as soon as the problem is sorted out...
> thanks in advance
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/anaconda-issues/issues/235#issuecomment-143412493>
> .
>



-- 
--Alain
619-884-5450

**Comment by goanpeca** [comment]
Hi, we are working on a new launcher which should address the issues presented by the previous version.

If you would like to try it please go to a terminal, remove the previous launcher, install the new version and launch it by typing:

conda remove launcher
conda install -c https://conda.anaconda.org/goanpeca/channel/dev launcher=2.0.0rc1
launcher
Let us know if the issues persist, or any other comments and suggestions

Cheers

**Comment by roblem** [comment]
On Fedora 22, I get the launcher window with only the background logo:

![launcher](https://cloud.githubusercontent.com/assets/1871086/10311657/aee1152e-6c15-11e5-83be-018a35a9d305.png)


and the following error:
```
[27565:1006/103032:ERROR:browser_main_loop.cc(170)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.
GLib-GIO-Message: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
[27565:1006/103033:ERROR:browser_main_loop.cc(216)] Gtk: GModule (/usr/lib64/gtk-2.0/2.10.0/immodules/im-ibus.so) initialization check failed: GLib version too old (micro mismatch)
[27565:1006/103033:ERROR:browser_main_loop.cc(216)] Gtk: Loading IM context type 'ibus' failed
libGL error: unable to load driver: nouveau_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: nouveau
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
[27565:1006/103033:INFO:CONSOLE(18894)] ""spawn: conda info --json"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (18894)
[27565:1006/103034:INFO:CONSOLE(19222)] ""Loading index ..."", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (19222)
[27565:1006/103034:INFO:CONSOLE(18894)] ""spawn: conda info --json --unsafe-channels"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (18894)
[27565:1006/103034:INFO:CONSOLE(19153)] ""Loading cache for channel https://repo.continuum.io/pkgs/pro/noarch/"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (19153)
[27565:1006/103034:INFO:CONSOLE(19155)] "" using cache file /home/username/anaconda/pkgs/cache/08159f17.json"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (19155)
[27565:1006/103034:INFO:CONSOLE(19153)] ""Loading cache for channel https://repo.continuum.io/pkgs/pro/linux-64/"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (19153)
[27565:1006/103034:INFO:CONSOLE(19155)] "" using cache file /home/username/anaconda/pkgs/cache/8db5e652.json"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (19155)
[27565:1006/103034:INFO:CONSOLE(19153)] ""Loading cache for channel https://repo.continuum.io/pkgs/free/noarch/"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (19153)
[27565:1006/103034:INFO:CONSOLE(19155)] "" using cache file /home/username/anaconda/pkgs/cache/9ca791dd.json"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (19155)
[27565:1006/103034:INFO:CONSOLE(19153)] ""Loading cache for channel https://repo.continuum.io/pkgs/free/linux-64/"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (19153)
[27565:1006/103034:INFO:CONSOLE(19155)] "" using cache file /home/username/anaconda/pkgs/cache/db552c1e.json"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (19155)
[27565:1006/103034:INFO:CONSOLE(19212)] ""Loaded conda index with 473 packages (7328 versions)"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (19212)
[27565:1006/103035:INFO:CONSOLE(17289)] ""ApplicationContent.render: rendering everything"", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (17289)
[27565:1006/103035:ERROR:nw_shell.cc(325)] TypeError: Cannot read property 'get' of undefined
    at React.createClass.getApps (file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js:17210:46)
    at React.createClass.render (file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js:17253:13)
    at ReactCompositeComponentMixin._renderValidatedComponentWithoutOwnerOrContext (/tmp/.org.chromium.Chromium.RXUD8k/node_modules/react/lib/ReactCompositeComponent.js:789:34)
    at ReactCompositeComponentMixin._renderValidatedComponent (/tmp/.org.chromium.Chromium.RXUD8k/node_modules/react/lib/ReactCompositeComponent.js:816:14)
    at ReactCompositeComponentMixin.mountComponent (/tmp/.org.chromium.Chromium.RXUD8k/node_modules/react/lib/ReactCompositeComponent.js:237:30)
    at Object.ReactReconciler.mountComponent (/tmp/.org.chromium.Chromium.RXUD8k/node_modules/react/lib/ReactReconciler.js:38:35)
    at ReactDOMComponent.ReactMultiChild.Mixin.mountChildren (/tmp/.org.chromium.Chromium.RXUD8k/node_modules/react/lib/ReactMultiChild.js:192:44)
    at ReactDOMComponent.Mixin._createContentMarkup (/tmp/.org.chromium.Chromium.RXUD8k/node_modules/react/lib/ReactDOMComponent.js:289:32)
    at ReactDOMComponent.Mixin.mountComponent (/tmp/.org.chromium.Chromium.RXUD8k/node_modules/react/lib/ReactDOMComponent.js:199:12)
    at Object.ReactReconciler.mountComponent (/tmp/.org.chromium.Chromium.RXUD8k/node_modules/react/lib/ReactReconciler.js:38:35)
[27565:1006/103035:INFO:CONSOLE(17210)] "Uncaught TypeError: Cannot read property 'get' of undefined", source: file:///tmp/.org.chromium.Chromium.RXUD8k/js/application.js (17210)
```

Also, I get the same error when trying to launch launcher using my workaround method outlined above.

**Comment by goanpeca** [comment]
Thanks for reporting wiil look into it!

**Comment by sandipank** [comment]
Getting same output as RobH...
Only a logo coming...
Not going inside...

On Tue, Oct 6, 2015 at 8:21 PM, Gonzalo Peña-Castellanos <
notifications@github.com> wrote:

> Thanks for reporting wiil look into it!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/anaconda-issues/issues/235#issuecomment-145881146>
> .
>



-- 
With Best Regards,
Sandipan Karmakar
Dept. of Industrial and Systems Engineering
Indian Institute of Technology, Kharagpur
West Bengal-721302
India
Alternate mail id: sandipankarmakar@hotmail.com
                         sandipank@iem.iitkgp.ernet.in
                 Web:
http://nitjsr.ac.in/departments/profile.php?user_id=MFE10
                          <09IM9405@iitkgp.ac.in>

**Comment by sandipank** [comment]
Hurray launcher is working...
Thanks for reporting...

On Wed, Oct 7, 2015 at 8:07 AM, Sandipan Karmakar <sandipank@gmail.com>
wrote:

> Getting same output as RobH...
> Only a logo coming...
> Not going inside...
>
> On Tue, Oct 6, 2015 at 8:21 PM, Gonzalo Peña-Castellanos <
> notifications@github.com> wrote:
>
>> Thanks for reporting wiil look into it!
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/ContinuumIO/anaconda-issues/issues/235#issuecomment-145881146>
>> .
>>
>
>
>
> --
> With Best Regards,
> Sandipan Karmakar
> Dept. of Industrial and Systems Engineering
> Indian Institute of Technology, Kharagpur
> West Bengal-721302
> India
> Alternate mail id: sandipankarmakar@hotmail.com
>                          sandipank@iem.iitkgp.ernet.in
>                  Web:
> http://nitjsr.ac.in/departments/profile.php?user_id=MFE10
>                           <09IM9405@iitkgp.ac.in>
>



-- 
With Best Regards,
Sandipan Karmakar
Dept. of Industrial and Systems Engineering
Indian Institute of Technology, Kharagpur
West Bengal-721302
India
Alternate mail id: sandipankarmakar@hotmail.com
                         sandipank@iem.iitkgp.ernet.in
                 Web:
http://nitjsr.ac.in/departments/profile.php?user_id=MFE10
                          <09IM9405@iitkgp.ac.in>

**Comment by goanpeca** [comment]
so it is working?

**Comment by sandipank** [comment]
Launcher is opening and showing install app but not moving after that...

On Wed, Oct 7, 2015 at 8:10 AM, Gonzalo Peña-Castellanos <
notifications@github.com> wrote:

> so it is working?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/anaconda-issues/issues/235#issuecomment-146062224>
> .
>



-- 
With Best Regards,
Sandipan Karmakar
Dept. of Industrial and Systems Engineering
Indian Institute of Technology, Kharagpur
West Bengal-721302
India
Alternate mail id: sandipankarmakar@hotmail.com
                         sandipank@iem.iitkgp.ernet.in
                 Web:
http://nitjsr.ac.in/departments/profile.php?user_id=MFE10
                          <09IM9405@iitkgp.ac.in>

**Comment by goanpeca** [comment]
Could you post a screenshot plus your system specs?

**Comment by cuent** [comment]
Hello!!
Did you find a solution when you can’t going inside? @sandipank.  

This the error message in my console:

```
31134:1010/221259:ERROR:browser_main_loop.cc(170)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.
ATTENTION: default value of option force_s3tc_enable overridden by environment.

(launcher:31134): IBUS-WARNING **: The owner of /home/cuent/.config/ibus/bus is not root!
[31134:1010/221306:INFO:CONSOLE(18894)] ""spawn: conda info --json"", source: file:///tmp/.org.chromium.Chromium.129XLK/js/application.js (18894)
[31134:1010/221307:INFO:CONSOLE(18957)] ""`conda info --json` failed with code 127:"", source: file:///tmp/.org.chromium.Chromium.129XLK/js/application.js (18957)
[31134:1010/221307:INFO:CONSOLE(18959)] ""bash: conda: command not found\n"", source: file:///tmp/.org.chromium.Chromium.129XLK/js/application.js (18959)
[31134:1010/221307:INFO:CONSOLE(18964)] ""Invalid JSON output from conda:"", source: file:///tmp/.org.chromium.Chromium.129XLK/js/application.js (18964)
[31134:1010/221307:INFO:CONSOLE(18965)] """", source: file:///tmp/.org.chromium.Chromium.129XLK/js/application.js (18965)
```

Cheers!!

**Comment by goanpeca** [comment]
@sandipank could you post a screenshot where it gets "stucked"?

Cheers

**Comment by sandipank** [comment]
Hi,
I had to reinstall my ubuntu on that day after launching the anaconda
launcher...
Wasted full 2 days again getting back into shape...
Then I left anaconda and custom installed python3 in ubuntu...
Thanks and regards

On Thu, Oct 15, 2015 at 7:21 PM, Gonzalo Peña-Castellanos <
notifications@github.com> wrote:

> @sandipank <https://github.com/sandipank> could you post a screenshot
> where it gets "stucked"?
>
> Cheers
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/anaconda-issues/issues/235#issuecomment-148393196>
> .
>



-- 
With Best Regards,
Sandipan Karmakar
Dept. of Industrial and Systems Engineering
Indian Institute of Technology, Kharagpur
West Bengal-721302
India
Alternate mail id: sandipankarmakar@hotmail.com
                         sandipank@iem.iitkgp.ernet.in
                 Web:
http://nitjsr.ac.in/departments/profile.php?user_id=MFE10
                          <09IM9405@iitkgp.ac.in>

**Comment by hjelmj** [comment]
Hi, any news on this? I just tried to start the Anaconda Launcher after installing it on Ubuntu 14.04.2 LTS
64-bit and if I search for it and click on the Anaconda Launcher icon I get the same error message
as reported previously in this thread:

Error: spawn ENOENT
    at exports._errnoException (util.js:742:11)
    at Process.ChildProcess._handle.onexit (child_process.js:1027:32)
    at child_process.js:1110:20
    at process._tickCallback (node.js:279:11)

Would be great to get an idea of when you think this might be resolved...I can get it to work fine if I start the launcher by issuing the "launcher" command in a terminal though. I am using Anaconda3 (Python 3.5), an otherwise excellent distribution.

**Comment by profversaggi** [comment]
Still a problem with a Fresh 14.05 LTS install .... launcher does not work ...

**Comment by sarajcev** [comment]
Still a problem with launcher 1.0 on Ubuntu 14.04 LTS.

**Comment by goanpeca** [comment]
Hi @sarajcev and @profversaggi.

Anaconda Launcher has been deprecated and there is now Anaconda Navigator replacing it.

`conda install anaconda-navigator`

**Comment by sarajcev** [comment]
Hi @goanpeca, thanks for the info!

**Comment by goanpeca** [comment]
:+1: -)

**Comment by KnoledgePro** [comment]
In my system (Ubuntu 14.04 w./ Unity), Anaconda Navigator needs to be started from command line.

**Comment by stephenakearns** [comment]
Hi @goanpeca is there a next step on this issue, or can we close this issue now?

**Comment by goanpeca** [comment]
I think we can close

**Comment by lloydbrodsky** [comment]
Yes, close

Sent from Yahoo Mail on Android 
 
  On Mon, Jun 27, 2016 at 9:26 PM, Gonzalo Peña-Castellanos<notifications@github.com> wrote:   
Closed #235.
